### PR TITLE
LPD-28410

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-28410

----

resend of https://github.com/liferay-commerce/liferay-portal/pull/5105

### CODE REVIEW

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+		CPDefinition cpDefinition = _cpDefinitionLocalService.getCPDefinition(
+			cpDefinitionVirtualSetting.getClassPK());
+
+		long fileEntryId = cpdVirtualSettingFileEntry.getFileEntryId();
+
+		BinaryFile binaryFile = multipartBody.getBinaryFile("file");
+
+		if (binaryFile != null) {
+			FileEntry fileEntry = FileEntryUtil.addFileEntry(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java:133]: probably the FileEntryUtil should be wrapped inside a specific wrapper for this ProductVirtualSettingsFileEntry

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+			fileEntryId = fileEntry.getFileEntryId();
+		}
+		else if (productVirtualSettingsFileEntry.getAttachment() != null) {
+			fileEntryId = FileEntryUtil.getFileEntryId(
+				productVirtualSettingsFileEntry.getAttachment(),
+				productVirtualSettingsFileEntry.getUrl(),
+				_uniqueFileNameProvider,
+				_serviceContextHelper.getServiceContext(
+					cpDefinition.getGroupId()));
+		}
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java:207]: this code is very similar to the one in the patch... is it possible to extract it to a private method?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+		throws Exception {
+
+		CPDefinitionVirtualSetting cpDefinitionVirtualSetting =
+			_cpDefinitionVirtualSettingService.getCPDefinitionVirtualSetting(
+				cpdVirtualSettingFileEntry.getCPDefinitionVirtualSettingId());
+
+		return HashMapBuilder.<String, Map<String, String>>put(
+			"delete",
+			ActionUtil.addAction(
+				ActionKeys.UPDATE, getClass(),
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsFileEntryResourceImpl.java:234]: there is no a DELETE permission for this resource?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+		CPDVirtualSettingFileEntry cpdVirtualSettingFileEntry =
+			_cpdVirtualSettingFileEntryService.getCPDVirtualSettingFileEntry(
+				id);
+
+		CPDefinitionVirtualSetting cpDefinitionVirtualSetting =
+			cpdVirtualSettingFileEntry.getCPDefinitionVirtualSetting();
+
+		_cpdVirtualSettingFileEntryService.deleteCPDVirtualSettingFileEntry(
+			CPInstance.class.getName(), cpDefinitionVirtualSetting.getClassPK(),
+			id);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java:69]: is not possible to delete directly the specific id?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+				cpDefinitionVirtualSetting.getClassPK(),
+				cpDefinitionVirtualSetting.getCPDefinitionVirtualSettingId(),
+				pagination.getStartPosition(), pagination.getEndPosition());
+
+		int totalItems =
+			cpDefinitionVirtualSetting.getCPDVirtualSettingFileEntriesCount();
+
+		return Page.of(
+			_toSkuVirtualSettingsFileEntries(cpdVirtualSettingFileEntries),
+			pagination, totalItems);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java:94]: can we write this like the previous class?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+
+	@Override
+	public SkuVirtualSettingsFileEntry patchSkuVirtualSettingsFileEntry(
+			Long id, MultipartBody multipartBody)
+		throws Exception {
+
+		CPDVirtualSettingFileEntry cpdVirtualSettingFileEntry =
+			_cpdVirtualSettingFileEntryService.getCPDVirtualSettingFileEntry(
+				id);
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsFileEntryResourceImpl.java:114]: the comments on the previous class should be applied also on this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+	extends BaseProductVirtualSettingsFileEntryResourceTestCase {
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_user = UserTestUtil.addOmniadminUser();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java:56]: I think is't not needed to have this global and can ben inline

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+			testGroup.getCompanyId(), testGroup.getGroupId(),
+			_user.getUserId());
+
+		_commerceCurrency = CommerceCurrencyTestUtil.addCommerceCurrency(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java:60]: I don't think this needs to be global

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+			testCompany.getCompanyId());
+
+		_commerceChannel = CommerceTestUtil.addCommerceChannel(
+			testCompany.getGroupId(), _commerceCurrency.getCode());
+
+		_commerceCatalog = CommerceTestUtil.addCommerceCatalog(
+			testCompany.getCompanyId(), testCompany.getGroupId(),
+			_user.getUserId(), _commerceCurrency.getCode());
+
+		_cpDefinition = CPTestUtil.addCPDefinitionFromCatalog(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java:70]: I don't think it's needed global

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+	@Override
+	protected void assertValid(
+			ProductVirtualSettingsFileEntry productVirtualSettingsFileEntry,
+			Map<String, File> multipartFiles)
+		throws Exception {
+
+		Assert.assertEquals(
+			new String(FileUtil.getBytes(multipartFiles.get("file"))),
+			_read(
+				"http://localhost:8080" +
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java:93]: are we sure this works on CI?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+	@DeleteAfterTestRun
+	private CommerceChannel _commerceChannel;
+
+	@DeleteAfterTestRun
+	private CommerceCurrency _commerceCurrency;
+
+	@DeleteAfterTestRun
+	private CPDefinition _cpDefinition;
+
+	@DeleteAfterTestRun
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductVirtualSettingsFileEntryResourceTest.java:225]: is it needed to delete them after every test?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/SkuVirtualSettingsFileEntryResourceTest.java [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Danny Situ
+ */
+@RunWith(Arquillian.class)
+public class SkuVirtualSettingsFileEntryResourceTest
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/SkuVirtualSettingsFileEntryResourceTest.java:48]: same comments of the previous class

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 13]
```diff
@@ -417,6 +424,32 @@ private Product _randomProductWithSku() throws Exception {
 		};
 	}
 
+	private Product _randomVirtualProduct() throws Exception {
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:427]: what about _randomProductVirtual? it's quite strange that we have `_testPostProductVirtual()` and `_randomVirtualProduct()`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 14]
```diff
@@ -467,6 +500,35 @@ else if (Objects.equals(fieldName, "width")) {
...
+		Product randomVirtualProduct = productResource.postProduct(
+			_randomVirtualProduct());
+
+		ProductVirtualSettings productVirtualSettings =
+			randomVirtualProduct.getProductVirtualSettings();
+
+		ProductVirtualSettingsFileEntry[] productVirtualSettingsFileEntries =
+			productVirtualSettings.getProductVirtualSettingsFileEntries();
+
+		Assert.assertTrue(productVirtualSettingsFileEntries.length == 1);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:529]: can we just check if the URL returned is empty or not?

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**